pkg/common/monitor.go

```diff
@@ -41,15 +44,39 @@ func FromScrapePath(exporter appsv1alpha1.Exporter) string {
 	return defaultScrapePath
 }
 
+func getPortNumberFromContainer(portName string, container *corev1.Container) string {
+	if container == nil || portName == "" {
+		return portName
+	}
+	for _, port := range container.Ports {
+		if port.Name == portName {
+			return strconv.Itoa(int(port.ContainerPort))
+		}
+	}
+	// Compatible with number ports.
+	return portName
+}
+
 func FromContainerPort(exporter Exporter, container *corev1.Container) string {
-	if exporter.ScrapePort != "" {
-		return exporter.ScrapePort
+	convertPort := func(port intstr.IntOrString) string {
+		switch {
+		case port.StrVal == "":
+			return getPortNumberFromContainer(exporter.TargetPort.StrVal, container)
+		case port.IntVal != 0:
+			return strconv.Itoa(int(exporter.TargetPort.IntVal))
+		default:
+			return ""
+		}
 	}
-	if container != nil && len(container.Ports) > 0 {
-		return container.Ports[0].Name
+
+	if exporter.ScrapePort != "" {
+		return getPortNumberFromContainer(exporter.ScrapePort, container)
 	}
 	if exporter.TargetPort != nil {
-		return exporter.TargetPort.String()
+		return convertPort(*exporter.TargetPort)
+	}
+	if container != nil && len(container.Ports) > 0 {
+		return strconv.Itoa(int(container.Ports[0].ContainerPort))
```

Comment:
what's this code used for?

Reply:
scrapePort is optional. If not specified, the port of the container will be used directly.